wordview/mwes/mwe.py

```diff
@@ -28,6 +28,10 @@ def __init__(
         """
         self.df = df
         self.text_column = text_column
+        if not mwe_types:
+            raise ValueError(f"mwe_types is empty.")
+        if not isinstance(mwe_types, list):
+            raise ValueError(f"mwe_types is not a list.")
```

Comment:
`TypeError`

Reply:
It doesn't raise a `TypeError` because:
> exception [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError)
> Raised when an operation or function is applied to an object of inappropriate type.

The problem in this case is two-fold:
1. We're auditing the arguments supplied by the client (hence not applying an operation to an object of the wrong type).
2. If the audit fails, it's because the user has supplied an incorrect `mwe_types` value.

> exception [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError):
> Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.

However, I can make it raise a TypeError if you prefer. It won't impact the code at all, just the logic for interpreting the argument auditing. Let me know what you decide and I can make the change.

Reply:
No, you are definitely right.

Reply:
> But then following the same argument, shouldn't then l:34 be a TypeError?

No, because as far as I can see, `mwe_types` _virtually_ never raises a `TypeError` anywhere. [Here](https://github.com/meghdadFar/wordview/blob/main/wordview/mwes/am.py#L58) and [here](https://github.com/meghdadFar/wordview/blob/main/wordview/mwes/am.py#L67) `mwe_types` is implicitly treated as an iterable, so if in the most likely error scenario a user passes just a string for a single MWE type (e.g. "NC"), `calculate_am()` will iterate over it character by character, setting the characters as the keys for the return object of `calculate_am()`.

What we're doing in `MWE.__init__()` is data validation, i.e. checking that the arguments passed by the user are correct, and not trying to apply an operation on the variable to which its type may or may not be appropriate. Given a lack of a *proper* data validation Exception in Python, `ValueError` would be the next best. I know the more dogmatic pythonistas would say that `TypeError` is  proper data validation Exception, but I disagree given type coercions (in general) and duck typing (python-specific).

To use a `TypeError` properly, we could do it like this, if e.g. user passes an `int` value for`mwe_types`.:
```
def calculate_am(count_data: dict, am: str, mwe_types: List[str]) -> Dict[str, Dict]:
    res = {}
    num_words = sum(count_data["WORDS"].values())
    if am == "pmi":
        try:
            for mt in mwe_types:
                ...
        except TypeError:
            logger.error("err == TypeError: 'int' object is not iterable")
```
The above implementation would go more in line with "Zen" of Python's "Explicit is better than implicit", than type-checking `mwe_types` distantly from where it is tried to be iterated over.

I will however change the error type in `MWE.__init__()` to `TypeError`, and just suggest that we move the `TypeError` to `calculate_am()` at a future iteration.